packages/core/src/internal/runSaga.js

```diff
@@ -73,9 +73,16 @@ export function runSaga(options, saga, ...args) {
     }
   }
 
+  let connectedChannel = channel
+  if (is.notUndef(dispatch)) {
+    if (process.env.NODE_ENV === 'development') {
+      check(dispatch, is.func, 'dispatch must be a function')
+    }
+    connectedChannel = channel.clone().lift(konst(wrapSagaDispatch(dispatch)))
```

Comment:
Lift the channel to be a connected-channel, so `env.channel` could be removed from `env`.